internal/client/categories/categories.go

```diff
@@ -0,0 +1,168 @@
+package categories
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/web-seven/provider-magento/apis/category/v1alpha1"
+
+	magento "github.com/web-seven/provider-magento/internal/client"
+)
+
+type Client struct {
+	magento.Client
+}
+
+func GetCategoryByName(c *magento.Client, categoryName string) (*v1alpha1.Category, error) {
+	path := fmt.Sprintf("%s/rest/all/V1/categories/list", c.BaseURL)
+
+	queryParams := map[string]string{
+		"searchCriteria[filterGroups][0][filters][0][field]": "name",
+		"searchCriteria[filterGroups][0][filters][0][value]": categoryName,
```

Comment:
@ca7alindev Bad idea, category identifier is ID. Clear is looks like not usable, but you have to keep logic of API, not think like abstract user of your API. Keep standards, that's what is waiting from you by user.